.circleci/config.yml

```diff
@@ -1,10 +1,21 @@
-version: 2
+version: 2.1
+commands:
+  publish:
+    parameters:
+      tag:
+        type: string
+    steps:
+        # Allows us to authenticate with the npm registry
+      - run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/.npmrc
```

Comment:
We should use scripts for everything that is more than single line

Reply:
Each of these individual steps is a script. Wrapping those ones up into a bigger script obfuscates how the release process works. I think this usage of CircleCI commands to not duplicate the npm publishing steps is really nice stuff.